### PR TITLE
Adding script to store status of base helm charts running on a cluster to cosmos DB

### PR DESCRIPTION
**Pull request by adusumillipraveen**
Also checks deprecation config (use the same Jenkins uses https://github.com/hmcts/cnp-jenkins-config/blob/master/deprecation-config.yml ) if the base chart is deprecated and adds it to cosmos.

Example stored:

```
{
    "id": "EF2F9ADD-1D30-44B8-9953-6C4D33474A25",
    "chartName": "pcq-consolidation-service",
    "namespace": "pcq",
    "clusterName": "cft-00-aks",
    "baseChart": "job",
    "baseChartVersion": "0.7.11",
    "isDeprecated": "false",
    "isError": "false",
    "_rid": "YWM1AMNZf64BAAAAAAAAAA==",
    "_self": "dbs/YWM1AA==/colls/YWM1AMNZf64=/docs/YWM1AMNZf64BAAAAAAAAAA==/",
    "_etag": "\"53004be4-0000-1100-0000-644792ee0000\"",
    "_attachments": "attachments/",
    "_ts": 1682412270
}
```

Example with deprecated base chart:

```
{
    "id": "D5733905-FFD0-4FB2-B44E-94C4B9AA9849",
    "chartName": "sptribs-dss-backend",
    "namespace": "sptribs",
    "clusterName": "cft-00-aks",
    "baseChart": "ccd",
    "baseChartVersion": "8.0.19",
    "isDeprecated": "true",
    "isError": "false",
    "_rid": "YWM1AMNZf64KAAAAAAAAAA==",
    "_self": "dbs/YWM1AA==/colls/YWM1AMNZf64=/docs/YWM1AMNZf64KAAAAAAAAAA==/",
    "_etag": "\"530056e4-0000-1100-0000-644793140000\"",
    "_attachments": "attachments/",
    "_ts": 1682412308
}
```

Tested in AAT-00 with sample job , run to completion.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: helm-base-versions/helm-chart-base-versions.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                    echo "$HR_NAME HR has base chart $DEPRECATED_CHART_NAME version $CURRENT_VERSION which is deprecated, please upgrade to at least ${DEPRECATED_CHART_VERSION}"
+                    break
+                fi
+          done
+          ./send-json-to-cosmos.sh $COSMOS_KEY "$CHART_NAME" "$NAMESPACE" "$CLUSTER_NAME" "$DEPRECATED_CHART_NAME" "$CURRENT_VERSION" "$IS_DEPRECATED" false
+        fi
+      done
+
+      rm -rf "/tmp/$CHART_NAME.tar.gz"
+      rm -rf /tmp/"$CHART_NAME"
```
Comment (thomast1906) [inline helm-base-versions/helm-chart-base-versions.sh:69]: can we not just rm the folder/chart rather than tar then this?

File: helm-base-versions/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+FROM alpine:3.17.3
```
Comment (timja) [inline helm-base-versions/Dockerfile:1]: Add renovate to this gets updated

File: helm-base-versions/helm-chart-base-versions.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+set -e
+
+[[ "$SCRIPT_DEBUG" == "true" ]] && set -x
+
+CLUSTER_NAME=${1:-$CLUSTER_NAME}
+ARTIFACT_URL=${2:-$ARTIFACT_URL}
+API_SERVER_URL=${3:-$API_SERVER_URL}
+COSMOS_KEY=${4:-$COSMOS_KEY}
```
Comment (timja) [inline helm-base-versions/helm-chart-base-versions.sh:9]: can we use managed identity auth please? or create a follow up ticket.
we aren't supposed to use key based auth and have previously told secops we only use AAD auth for our team and cosmos

File: helm-base-versions/example-job.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        serviceAccountName: helm-chart-base-versions-sa
+        containers:
+        - name: helm-chart-base-versions
+          image: hmctspublic.azurecr.io/helm-base-versions:latest
+          imagePullPolicy: Always
+          env:
+            - name: CLUSTER_NAME
+              value: "cft-00-aks"
+            - name: COSMOS_KEY
+              value: "<your key in here>"
```
Comment (timja) [inline helm-base-versions/example-job.yaml:24]: shouldn't really use this as an env var should be a secret

File: helm-base-versions/helm-chart-base-versions.sh [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    if [ -n "$CHART_PATH" ] && [ "$CHART_PATH" != "null" ]; then
+
+      #Chart name can be different to HR name in some cases where multiple HRs use same charts like camunda, logstash
+      CHART_NAME=$(echo "$CHART_PATH" | cut -f4 -d"/" | sed 's/\(.*\)-.*/\1/')
+
+      curl -Ls "$ARTIFACT_URL/$CHART_PATH" -o "/tmp/$CHART_NAME.tar.gz"
+      tar -xf "/tmp/$CHART_NAME.tar.gz" -C /tmp/
+
+      for DEPRECATED_CHART_NAME in $( echo "${DEPRECATION_CONFIG}" | jq -r 'keys | .[]' ); do
+        #echo "checking $DEPRECATED_CHART_NAME"
```
Comment (timja) [inline helm-base-versions/helm-chart-base-versions.sh:49]: is this no longer useful? seems like it could be helpful

File: helm-base-versions/send-json-to-cosmos.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+RESOURCE_LINK="$RESOURCE_ID/$RESOURCE_TYPE"
+COSMOS_URL="https://pipeline-metrics.documents.azure.com/$RESOURCE_LINK"
+NOW=$(TZ=GMT date '+%a, %d %b %Y %T %Z')
+
+SIGNATURE="$(printf "%s" "$VERB\n$RESOURCE_TYPE\n$RESOURCE_ID\n$NOW" | tr '[A-Z]' '[a-z]')\n\n"
+HEX_KEY=$(printf "$COSMOS_KEY" | base64 --decode | hexdump -v -e '/1 "%02x"')
+hashedSignature=$(printf "$SIGNATURE" | openssl dgst -sha256 -mac hmac -macopt hexkey:$HEX_KEY -binary | base64)
+AUTH_STRING="type=master&ver=1.0&sig=$hashedSignature"
+URL_ENCODED_STRING=$(printf "$AUTH_STRING" | sed 's/=/%3d/g' | sed 's/&/%26/g' | sed 's/+/%2b/g' | sed 's/\//%2f/g')
+
```
Comment (timja) [inline helm-base-versions/send-json-to-cosmos.sh:26]: I suspect it would be safer to use an SDK via node.js or python.

there's a risk here that the bash tools won't handle some edge cases that the SDKs / a higher level language have automated test coverage for and would be easier to understand.

File: .github/renovate.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": [
+    "github>hmcts/.github//renovate/global"
```
Comment (timja) [inline .github/renovate.json:4]: can probably automerge here? or do you want test coverage first?

File: helm-base-versions/send-json-to-cosmos.sh [hunk 8]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
```
Comment (timja) [inline helm-base-versions/send-json-to-cosmos.sh:1]: It would be good to have some test coverage for this script ideally I think

Comment (adusumillipraveen) [reply helm-base-versions/send-json-to-cosmos.sh:1]: checking if its easy to move over to any sdk

**Review by timja — APPROVED** [review]
